youths/schema.py

```diff
@@ -0,0 +1,216 @@
+            try:
+                profile = Profile.objects.get(user=info.context.user)
+            except Profile.DoesNotExist:
+                raise GraphQLError(
+                    "No profile found. Please create a profile before creating a youth profile!"
+                )
+
+        youth_profile = YouthProfile.objects.get(profile=profile)
+        for field, value in input_data.items():
+            setattr(youth_profile, field, value)
+        youth_profile.save()
+
+        if resend_request_notification:
+            youth_profile.approval_token = create_approval_token()
+            mail_approval_link(youth_profile)
+            youth_profile.approval_notification_sent = datetime.now()
+            youth_profile.save()
+
+        return UpdateYouthProfile(youth_profile=youth_profile)
+
+
+class ApproveYouthProfileInput(graphene.InputObjectType):
+    diabetes = graphene.Boolean()
+    epilepsy = graphene.Boolean()
+    heart_disease = graphene.Boolean()
+    extra_illnesses_info = graphene.String()
+    serious_allergies = graphene.Boolean()
+    allergies = graphene.String()
+    photo_usage_approved = graphene.Boolean()
+
+
+class ApproveYouthProfile(graphene.Mutation):
+    class Arguments:
+        approval_token = graphene.String(required=True)
+        approval_data = ApproveYouthProfileInput(required=True)
+
+    youth_profile = graphene.Field(YouthProfileType)
+
+    def mutate(self, info, **kwargs):
+        youth_data = kwargs.pop("approval_data")
+        token = kwargs.get("approval_token")
+
+        try:
+            data = signing.loads(token)
+            youth_profile = YouthProfile.objects.get(
+                pk=data.get("id"), approval_token=data.get("approval_token")
+            )
+        except (ObjectDoesNotExist, signing.BadSignature):
+            raise GraphQLError("Invalid approval token")
+
+        for field, value in youth_data.items():
+            setattr(youth_profile, field, value)
+
+        youth_profile.approved_time = datetime.now()
+        youth_profile.approval_token = ""  # invalidate
+        youth_profile.save()
+        send_notification(
+            email=youth_profile.profile.email,
+            notification_type=NotificationType.YOUTH_PROFILE_CONFIRMED.value,
+            context={"youth_profile": youth_profile},
+        )
+        return ApproveYouthProfile(youth_profile=youth_profile)
+
+
+class Query(graphene.ObjectType):
+    youth_profile = graphene.Field(YouthProfileType, profile_id=graphene.ID())
+
+    # TODO should we hide some fields (e.g. gender) from the approver?
+    youth_profile_by_approval_token = graphene.Field(
+        YouthProfileType, token=graphene.String()
+    )
+
+    @login_required
+    def resolve_youth_profile(self, info, **kwargs):
+        if (
+            info.context.user.is_superuser
+        ):  # FIXME should be something like user.is_staff_or_more?
+            return YouthProfile.objects.get(profile_id=kwargs.get("profile_id"))
```

Comment:
probably also needs to raise `GraphQLError` if no `Profile` exists

Reply:
Indeed. Missed it here, thanks!